kernel/src/syscall.rs

```diff
@@ -522,111 +552,197 @@ impl SyscallReturn {
     /// Encode the system call return value into 4 registers, following the
     /// encoding specified in TRD104. Architectures which do not follow TRD104
     /// are free to define their own encoding.
+    /// TODO: deprecate in favour of the more general one
```

Comment:
It seems perfectly fine for this kernel-supplied implementation to only cover a narrow set of platforms. We deliberately allow downstream users or individual architectures to use their own implementations. The comment above seems fine as is now, and once TRD 104 is updated to reflect these changes we should update the above comment instead of adding this TODO.

Reply:
Ignore my comment above, we instead agreed on renaming this to be TRD104-specific (which #4229 implements).